lib/uaa/token_issuer.rb

```diff
@@ -74,8 +94,18 @@ def request_token(params)
       params[:scope] = Util.strlist(scope)
     end
     headers = {'content-type' => FORM_UTF8, 'accept' => JSON_UTF8}
-    if @basic_auth
-      headers['authorization'] = Http.basic_auth(@client_id, @client_secret)
+    if @client_auth_method == 'client_secret_basic' && @client_secret && @client_id
+      if @basic_auth
+        headers['authorization'] = Http.basic_auth(@client_id, @client_secret)
+      else
+        headers['X-CF-ENCODED-CREDENTIALS'] = 'true'
+        headers['authorization'] = Http.basic_auth(CGI.escape(@client_id), CGI.escape(@client_secret))
+      end
+    elsif @client_auth_method == 'client_secret_post' && @client_secret && @client_id
```

Comment:
Is enabling the option of "client_secret_basic" vs. "client_secret_post" relevant to the main purpose of this PR (PKCE)? Could this be a separate PR, or at least a separate commit?

Reply:
done